src/controller/event/handler/webhook/artifact/retention.go

```diff
@@ -0,0 +1,144 @@
+package artifact
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/goharbor/harbor/src/core/api"
+
+	"github.com/goharbor/harbor/src/controller/event"
+	"github.com/goharbor/harbor/src/controller/event/handler/util"
+	"github.com/goharbor/harbor/src/core/config"
+	"github.com/goharbor/harbor/src/lib/log"
+	"github.com/goharbor/harbor/src/pkg/notification"
+	"github.com/goharbor/harbor/src/pkg/notifier/model"
+)
+
+// RetentionHandler preprocess tag retention event data
+type RetentionHandler struct {
+}
+
+// Handle ...
+func (r *RetentionHandler) Handle(value interface{}) error {
+	if !config.NotificationEnable() {
```

Comment:
the logic is handled in the SendHookWithPolicies

Reply:
> the logic is handled in the SendHookWithPolicies

Fast return. while the value is false, we don't need to do the redundant parsing for event.

Reply:
I agree with the return early. 

But, if we do want to do `fast return`, can we define it at one place at somewhere instead of at any of handler? In case someone who doesn't know this and introduces a new handler.

Reply:
now the model is  notification->event build->event publish->trigger webhook(handler sending webhook).   There are two parts: 1. event build&publish; 2. webhook sending.  The handler is the entrypoint to process webhook, so we have to judge webhook enable in the handler.